src/macports1.0/macports.tcl

```diff
@@ -711,7 +713,8 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
             set os_subplatform macosx
             if {[file executable /usr/bin/sw_vers]} {
                 try -pass_signal {
-                    set macosx_version [exec /usr/bin/sw_vers -productVersion | cut -f1,2 -d.]
+                    set macosx_full_version [exec /usr/bin/sw_vers -productVersion]
+                    set macosx_version [join [lrange [split $macosx_full_version "."] 0 end-1] "."]
```

Comment:
Should macOS ever start using four-tupled version numbers, this is a change of behavior. I'd prefer

```suggestion
                    set macosx_version [join [lrange [split $macosx_full_version "."] 0 1] "."]
```